core/packet.c

```diff
@@ -109,31 +110,46 @@ static coap_status_t handle_request(lwm2m_context_t * contextP,
     coap_status_t result = NOT_FOUND_4_04;
 
     uriP = lwm2m_decode_uri(message->uri_path);
-    if (uriP == NULL) return BAD_REQUEST_4_00;
-
-    switch(uriP->flag & LWM2M_URI_MASK_TYPE)
-    {
+    if (uriP == NULL) {
```

Comment:
uriP can be NULL is case of invalid URI, thus a Delete on /2/toto would be interpreted as a Delete on /.
You should keep a special flag for / and return it when the uriPath is empty.